core/src/consensus/tendermint/engine.rs

```diff
@@ -134,6 +134,49 @@ impl ConsensusEngine for Tendermint {
 
     fn stop(&self) {}
 
+    /// Block transformation functions, before the transactions.
+    fn on_open_block(&self, block: &mut ExecutedBlock) -> Result<(), Error> {
+        let client = self.client().ok_or(EngineError::CannotOpenBlock)?;
+
+        let block_number = block.header().number();
+        let era = client.term_common_params(block_number.into()).map_or(0, |p| p.era());
+        let metadata = block.state().metadata()?.expect("Metadata must exist");
+        if block_number == metadata.last_term_finished_block_num() + 1 {
+            match era {
+                0 => {}
+                1 => {
+                    let rewards = stake::v1::drain_current_rewards(block.state_mut())?;
+                    let start_of_the_current_term = block_number;
+                    let start_of_the_previous_term = {
+                        let end_of_the_two_level_previous_term = client
+                            .last_term_finished_block_num((metadata.last_term_finished_block_num() - 1).into())
+                            .unwrap();
+
+                        end_of_the_two_level_previous_term + 1
+                    };
+
+                    let banned = stake::Banned::load_from_state(block.state())?;
+                    let start_of_the_current_term_header =
```

Comment:
Does the header contains seal here?

Reply:
I checked the related parts, and I'm pretty sure that the seal exists at this point.